Documentation/upgrades/upgrade_3_4.md

```diff
@@ -0,0 +1,163 @@
+## Upgrade etcd from 3.3 to 3.4
+
+In the general case, upgrading from etcd 3.3 to 3.4 can be a zero-downtime, rolling upgrade:
+ - one by one, stop the etcd v3.3 processes and replace them with etcd v3.4 processes
+ - after running all v3.4 processes, new features in v3.4 are available to the cluster
+
+Before [starting an upgrade](#upgrade-procedure), read through the rest of this guide to prepare.
+
+### Upgrade checklists
+
+Highlighted breaking changes in 3.4.
+
+#### Change in `etcdserver.ServerConfig` struct
+
+`etcdserver.ServerConfig` has added a new field `HostWhitelist` and its default value is empty. If empty and etcd server does not enable TLS, all HTTP client requests will be rejected. This is to prevent [DNS Rebinding](https://en.wikipedia.org/wiki/DNS_rebinding) [attacks](https://bugs.chromium.org/p/project-zero/issues/detail?id=1447). Set it to `"*"` to allow all hostnames.
+
+Before and after (e.g. [k8s.io/kubernetes/test/e2e_node/services/etcd.go](https://github.com/kubernetes/kubernetes/blob/release-1.10/test/e2e_node/services/etcd.go#L78-L81))
```

Comment:
@jpbetz FYI, this will be breaking change for [Kubernetes testing with etcd v3.4](https://github.com/kubernetes/kubernetes/blob/release-1.10/test/e2e_node/services/etcd.go#L78-L81).

Reply:
Can we avoid this by having an empty `HostWhiteList` mean "allow all" instead of `[]string{"*"}`.

Reply:
@ericchiang Good idea. Then, we won't have to worry about breaking changes. Will change.